src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/AndroidBrokerProxy.cs

```diff
@@ -226,21 +238,52 @@ public Intent GetIntentForBrokerActivity(AuthenticationRequest request, Activity
                 // logic
                 if (intent != null)
                 {
+                    _logger.Info("BrokerProxy: Intent created from BundleResult is not null. ");
                     intent.PutExtra(BrokerConstants.BrokerRequest, BrokerConstants.BrokerRequest);
                 }
+                else
+                {
+                    _logger.Info("BrokerProxy: Intent created from BundleResult is null. ");
+                }
             }
-            catch (AdalException)
+            catch (AdalException ex)
             {
-                throw;
+                _logger.ErrorPii(ex);
+                throw new AdalException(
```

Comment:
any reason this can't continue to be a throw? is an unknown error the right thing? why? If yes, perhaps a comment or a method that throws this message which has a meaningful name?

Reply:
i put back the throw, but kept the logging.

---
In reply to: [288388935](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r288388935) [](ancestors = 288388935)